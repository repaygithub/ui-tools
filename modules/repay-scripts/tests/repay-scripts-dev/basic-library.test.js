const TestSetup = require('../test-setup')

beforeEach(() => jest.setTimeout(1000 * 60 * 5))

const sleep = seconds => {
  return new Promise(resolve => {
    setTimeout(resolve, seconds * 1000)
  })
}

describe('@repay/repay-scripts', () => {
  let test

  beforeEach(() => {
    test = new TestSetup('basic-library')
  })

  afterEach(async () => {
    await test.cleanup()
    test = undefined
  })

  it('should watch a basic library for local development', async () => {
    await test.setup()
    const pkg = JSON.parse(await test.readFile('package.json'))
    test.spawn('yarn', ['repay-scripts', 'dev', '--lib', 'index.js'])
    await test.waitForText(pkg.module)
    await sleep(0.5)
    expect(await test.readFile(pkg.module)).toMatchSnapshot()
    await test.updateFile('app.js')
    await test.waitForText(pkg.module, 2)
    await sleep(0.5)
    expect(await test.readFile(pkg.module)).toMatchSnapshot()
  })
})
