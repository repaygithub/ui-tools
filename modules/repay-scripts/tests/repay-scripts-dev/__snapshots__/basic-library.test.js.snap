// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@repay/repay-scripts should watch a basic library for local development 1`] = `
"import React from 'react';

function App(props) {
  return React.createElement(\\"div\\", {
    className: \\"AppWrapper\\"
  }, props.children);
}

export { App };
//# sourceMappingURL=build.esm.js.map
"
`;

exports[`@repay/repay-scripts should watch a basic library for local development 2`] = `
"import React from 'react';

function App(props) {
  return React.createElement(\\"div\\", {
    className: \\"AppWrapper Updated\\"
  }, props.children);
}

export { App };
//# sourceMappingURL=build.esm.js.map
"
`;
